querybook/server/lib/ai_assistant/base_ai_assistant.py

```diff
@@ -50,17 +62,37 @@ def name(self) -> str:
     def set_config(self, config: dict):
         self._config = config
 
+    def catch_error(func):
+        @functools.wraps(func)
+        def wrapper(self, *args, **kwargs):
+            try:
+                return func(self, *args, **kwargs)
+            except Exception as e:
+                LOG.error(e, exc_info=True)
+                callback_handler = kwargs.get("callback_handler")
```

Comment:
callback_handler is required, can you make it more explicit (not just in except)
it would be useful to note that callback_handler must be passed as kwargs not args

Reply:
actually callback_handler is not always required, only when it's a streaming function. and catch_error will also handle non streaming functions as well. will update